config/sample-idex.cfg

```diff
@@ -90,7 +114,149 @@ gcode:
+        {% set newX = params.X_ADJUST|float + oldX %}
+        SAVE_VARIABLE VARIABLE=xoffset VALUE={ newX|float }
+    {% endif %}
+
+    {% if params.Y_ADJUST is defined %}
+        {% set newY = params.Y_ADJUST|float + oldY %}
+        SAVE_VARIABLE VARIABLE=yoffset VALUE={ newY|float }
+    {% endif %}
+
+    {% if params.Z_ADJUST is defined %}
+        {% set newZ = params.Z_ADJUST|float + oldZ %}
+        SAVE_VARIABLE VARIABLE=yoffset VALUE={ newZ|float }
+    {% endif %}
+
+[gcode_macro calibrate_separation]
+gcode:
+    G28
+    G90
+    M83
+    T0 ; test T0
+    G1 X120 Y150 Z.2 F4800
+    G1 Y75 E10
+    T1 ; test T1
+    G1 X120 Y0 Z.2
+    G1 Y75 E10
+
+# The following is for if your carraiges have their own part cooling fans.
+# Since this macro overrides M106, you cannot also have a [fan] section defined.
+# If your printer only has one part cooling fan, don't copy this section
+[fan_generic part_fan_0]
+pin: PE5
+
+[fan_generic part_fan_1]
+pin: PD13
+
+# For completeness, you can add a [fan] section with an unused pin
+[fan]
+pin: rpi:gpio20
+
+[gcode_macro M106]
+# Only rename_existing if you have a sacrificial [fan] section
+rename_existing: M106.1
+# The variable that controls fan speed swopping if not specifying P parameter
+# -1 means the control is disabled, a value of 0-1 is the requested fan speed.
+# Access via {printer["gcode_macro M106"].swap_speed}
+variable_swap_speed: -1
+gcode:
+    {% set s = [[params.S|default(255)|int, 255]|min, 0]|max %}
+    {% set p = params.P|default(-1)|int %}
+    {% set speed = s / 255 %}
+
+    # Set speed to -1 by default
+    SET_GCODE_VARIABLE MACRO=M106 VARIABLE=swap_speed VALUE=-1
+
+    {% if p == -1 %}
+      # Set current active extruder fan
+      {% if speed == 0 %}
+        # Always turn off al fans if S0 is specified without a specific fan
+        SET_FAN_SPEED FAN=fan_extruder SPEED=0
+        SET_FAN_SPEED FAN=fan_extruder1 SPEED=0
+      {% else %}
+        # Opt into fan speed swop control
+        SET_GCODE_VARIABLE MACRO=M106 VARIABLE=swap_speed VALUE={speed}
+        SET_FAN_SPEED FAN=fan_{printer.toolhead.extruder} SPEED={speed}
+      {% endif %}
+    {% else %}
+      # Set specified active extruder fan
+      {% if p == 0 %}
+        SET_FAN_SPEED FAN=fan_extruder SPEED={speed}
+      {% else %}
+        SET_FAN_SPEED FAN=fan_extruder1 SPEED={speed}
+      {% endif %}
+    {% endif %}
+
+    # Update core Klipper's fan speed
+    # Only do this if you have a sacrificial [fan] section
+    M106.1 S{s}
+
+[gcode_macro M107]
+rename_existing: M107.1
+gcode:
+    {% set p = params.P|default(-1)|int %}
+    M106 S0 P{p}
+
+[gcode_macro M107]
+rename_existing: M107.1
+gcode:
+    {% set p = params.P|default(-1)|int %}
+    M106 S0 P{p}
+
+[save_variables]
+filename: ~/klipper_config/variables.klip
+# this is used for saving and restoring the idex offsets
+# create the file and insert the folloring content (without the #) to start
+#
+# [Variables]
+# xoffset = 0.0
```

Comment:
I didn't try it myself, but would it be possible to use `|default(0.0)` at the place of their use? Or `is defined`. So that the user does not have to manually add this section to the save variable file. But maybe it doesn't work that way.